packages/date-helpers/src/date-helpers.ts

```diff
@@ -0,0 +1,194 @@
+      DateHelpers.usLocaleDateFormat,
+      DateHelpers.isoDateFormat,
+      'yyyy/MM/dd',
+      'M/d/yyyy',
+      'MMM d, yyyy',
+      'MMM yyyy',
+      DateHelpers.usLocaleTimeFormat,
+    ];
+
+    possibleFormats.some((pattern) => {
+      output = parse(input, pattern, new Date());
+      return isValid(output);
+    });
+
+    if (!isValid(output)) {
+      withScope((scope: Scope) => {
+        scope.setLevel('error');
+        scope.setExtras({
+          input,
+        });
+        captureException(
+          new InvalidArgumentException(
+            `Invalid Date: unable to parse date string - ${input}`,
+          ),
+        );
+      });
+    }
+
+    return output;
+  }
+
+  /**
+   * Formats a date as an ISO date string.
+   *
+   * @example
+   * ```typescript
+   * const isoDateString = DateHelpers.formatIsoDate(new Date(2023, 0, 31));
+   * // Result: "2023-01-31"
+   *
+   * // or
+   *
+   * const isoDateString = DateHelpers.formatIsoDate("2025/10/20");
+   * // Result: "2025-10-20"
+   * ```
+   *
+   * @param dateSource - The date to format, can be a Date object or a string.
+   * @returns The formatted date string.
+   */
+  public static formatIsoDate(dateSource: DateType): string {
+    // Do nothing with empty strings
+    if (dateSource === '') {
+      return dateSource;
+    }
+
+    return formatISO(getParsedDate(dateSource), {
+      representation: 'date',
+    });
+  }
+
+  /**
+   * Formats a date as a US locale date string.
+   *
+   * @example
+   * ```typescript
+   * const usDateString = DateHelpers.formatUsLocaleDate(new Date(2023, 0, 31));
+   * // Result: "01/31/2023"
+   *
+   * // or
+   *
+   * const usDateString = DateHelpers.formatUsLocaleDate("2025/10/20");
+   * // Result: "10/20/2025"
+   * ```
+   *
+   * @param dateSource - The date to format, can be a Date object or a string.
+   * @param dateFormat - Optional, a custom date format string.
+   * @returns The formatted date string.
+   */
+  public static formatUsLocaleDate(
+    dateSource: DateType,
+    dateFormat?: string,
+  ): string {
+    // Do nothing with empty strings
+    if (dateSource === '') {
+      return dateSource;
+    }
+
+    if (dateFormat) {
+      return format(getParsedDate(dateSource), dateFormat);
+    }
+
+    return format(getParsedDate(dateSource), DateHelpers.usLocaleDateFormat);
+  }
+
+  /**
+   * Formats a date as a US locale time string.
+   *
+   * @example
+   * ```typescript
+   * const usTimeString = DateHelpers.formatUsLocaleTime(new Date(2023, 0, 31, 12, 0, 0));
+   * // Result: "01/31/2023 12:00:00"
+   *
+   * // or
+   *
+   * const usTimeString = DateHelpers.formatUsLocaleTime("2025/10/20 12:00:00");
+   * // Result: "10/20/2025 12:00:00"
+   * ```
+   *
+   * @param dateSource - The date to format, can be a Date object or a string.
+   * @param dateFormat - Optional, a custom date format string.
+   * @returns The formatted time string.
+   */
+  public static formatUsLocaleTime(
```

Comment:
```suggestion
  public static formatUsLocaleDateTime(
```

Reply:
formatDateToUSLocaleDateTimeString ref: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toLocaleTimeString

Reply:
er... `formatDateToXxx` is also not a good practice. our source param is not Date.   |o|